### PR TITLE
Improves description of optimising parallel IO on ARCHER2

### PR DESCRIPTION
**Pull request by aturner-epcc**

### CODE REVIEW

File: docs/user-guide/io.md [hunk 1]
```diff
@@ -106,6 +106,88 @@ system provides POSIX semantics (changes on one node are immediately
...
+- Environment variables:
+    * `export FI_OFI_RXM_SAR_LIMIT=64K`
+    * `export MPICH_MPIIO_HINTS="*:cray_cb_write_lock_mode=2,*:cray_cb_nodes_multiplier=4”`
+- MPI transport protocol choice: UCX, if possible
+
+### Summary: typical I/O performance on ARCHER2
+
+#### File-Per-Process (FPP)
+
+We regularly run tests of FPP write performance on ARCHER2 /work Lustre file
```
Comment (juanfrh) [inline docs/user-guide/io.md:145]: I'd use `/work` (with backticks).

File: docs/user-guide/io.md [hunk 2]
```diff
@@ -106,6 +106,88 @@ system provides POSIX semantics (changes on one node are immediately
...
+    * `export FI_OFI_RXM_SAR_LIMIT=64K`
+    * `export MPICH_MPIIO_HINTS="*:cray_cb_write_lock_mode=2,*:cray_cb_nodes_multiplier=4”`
+- MPI transport protocol choice: UCX, if possible
+
+### Summary: typical I/O performance on ARCHER2
+
+#### File-Per-Process (FPP)
+
+We regularly run tests of FPP write performance on ARCHER2 /work Lustre file 
+systems using the [benchio]() software in the following configuration:
```
Comment (juanfrh) [inline docs/user-guide/io.md:146]: The _benchio_ link does not point to any place.

File: docs/user-guide/io.md [hunk 3]
```diff
@@ -106,6 +106,88 @@ system provides POSIX semantics (changes on one node are immediately
...
+    - Lower quartile: 135 GiB/s
+- HDD-based file system:
+    - Maximum: 170 GiB/s
+    - Upper quartile: 160 GiB/s
+    - Median: 90 GiB/s
+    - Lower quartile: 30 GiB/s
+
+#### Single Shared File with collective writes (SSF)
+
+We regularly run tests of FPP write performance on ARCHER2 /work Lustre file
```
Comment (juanfrh) [inline docs/user-guide/io.md:169]: I'd use `/work` (with backticks).

File: docs/user-guide/io.md [hunk 4]
```diff
@@ -106,6 +106,88 @@ system provides POSIX semantics (changes on one node are immediately
...
+- HDD-based file system:
+    - Maximum: 170 GiB/s
+    - Upper quartile: 160 GiB/s
+    - Median: 90 GiB/s
+    - Lower quartile: 30 GiB/s
+
+#### Single Shared File with collective writes (SSF)
+
+We regularly run tests of FPP write performance on ARCHER2 /work Lustre file 
+systems using the [benchio]() software in the following configuration:
```
Comment (juanfrh) [inline docs/user-guide/io.md:170]: The `benchio` link does not point to any place.

**Review by juanfrh — COMMENTED** [review]
Just some minor comments. The rest looks good to me.

File: docs/user-guide/io.md [hunk 5]
```diff
@@ -167,32 +249,26 @@ For example, to set a stripe size of 4 MiB for the existing directory
 
     auser@ln03:~> lfs setstripe -S 4m -c -1 resdir/
 
-### Recommended ARCHER2 I/O settings
+### Environement variables
```
Comment (kevinstratford) [inline docs/user-guide/io.md:252]: Typo.

File: docs/user-guide/io.md [hunk 6]
```diff
@@ -167,32 +249,26 @@ For example, to set a stripe size of 4 MiB for the existing directory
...
 
+These environment variables typically only have an impact for the case
```
Comment (kevinstratford) [inline docs/user-guide/io.md:254]: "These"?

File: docs/user-guide/io.md (hunk 5 above)
Comment (juanfrh) [reply docs/user-guide/io.md]: Fixed.

File: docs/user-guide/io.md (hunk 6 above)
Comment (juanfrh) [reply docs/user-guide/io.md]: I've changed some wording. Please could you have a look at it? @kevinstratford

**Review by juanfrh — APPROVED** [review]

**Pull request merged**